torchvision/models/__init__.py

```diff
@@ -13,9 +15,23 @@
 from .vgg import *
 from .vision_transformer import *
 from .swin_transformer import *
-from . import detection
 from . import optical_flow
 from . import quantization
 from . import segmentation
 from . import video
 from ._api import get_weight
+
+
+def __getattr__(name):
```

Comment:
This adds the `detection` attribute lazily. If we imported it like the others (e.g. `quantization` above), users would get the "beta detection warning" even when just importing `torchvision.models`. See more details about `__getattr__` at https://peps.python.org/pep-0562/

Note: the detection attribute is needed when doing e.g.
```py
import torchvision.models as M
M.detection  # <- directly accessing the attribute
```

but in a lot of patterns this gets bypassed, typically when doing

```py
from torchvision.models.detection import ssd
```
or even just

```py
import torchvision.models.detection
```